src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,42 @@
+import { useRef, useState } from "react";
+import ModalBasic from "./ModalBasic/ModalBasic";
+import styles from "./Modal.module.scss";
+import { useOnClickOutside } from "usehooks-ts";
+
+interface Modal {
+  name: string;
+  subtitle?: string;
+  shareButtons?: boolean;
```

Comment:
boolean 데이터 타입은 변수명으로 의미를 조금 더 드러내주면 좋을 것 같아요~
이런 느낌이 좋을 것 같네요!
```suggestion
  hasShareButtons?: boolean;
```